apps/src/aichat/aichatApi.ts

```diff
@@ -101,3 +102,28 @@ export async function postAichatCheckSafety(
 
   return await response.json();
 }
+
+/**
+ * This function sends a GET request to the aichat student chat history backend controller, then returns
+ * a list of chat events if successful.
+ */
+export async function getStudentChatHistory(
+  studentUserId: number,
+  levelId: number,
+  scriptId: number | null,
+  scriptLevelId: number | undefined
+): Promise<ChatEvent[]> {
+  const params: Record<string, string> = {
+    studentUserId: studentUserId.toString(),
+    levelId: levelId.toString(),
+    scriptId: scriptId?.toString() || '',
+  };
+  if (scriptLevelId) {
+    params.scriptLevelId = scriptLevelId.toString();
+  }
+  const urlParams = new URLSearchParams(params);
+  const response = await HttpClient.fetchJson<ChatEvent[]>(
+    STUDENT_CHAT_HISTORY_URL + '?' + new URLSearchParams(urlParams.toString())
```

Comment:
Why do we need to create two `URLSearchParams`? Could we just do `` `${STUDENT_CHAT_HISTORY_URL}?${new URLSearchParams(params)}` ``

Reply:
Oh yeah! Think that's just an oversight. Will update.